### PR TITLE
Security issue

### PR DESCRIPTION
**Pull request by Rupeshiya**
---
name: Pull Request template
about: Describe the PR.
title: ''
labels: ''
assignees: ''

---

### **Problem**
Github Issue Number: #176 
Fix security flaws

### Fixed:
- [X] DOS attack
- [X] XSS attack 
- [X] Rate limiter
- [X] CSRF token attack 
- [X] Clickjacking attack
- [X] protect against HTTP Parameter Pollution attacks using HPP
- [X] Hide password from client-side
- [X] Hide user tokens

### **Type of Change**
[ ] Bug fix
[ ] New Feature
[ ] Development of UI/UX prototypes
[&check;] Small refactor
[ ] Change in Documentation

### **Checklist**
[&check;] My code follows the same style as the codebase
[ ] My Code change requires a change in documentation
[ ] I have updated the Readme accordingly
[&check;] I made PR against **development branch**
[&check;] I have run the test cases locally and it's passing.
[ ] I have squashed my commits

### CODE REVIEW

File: app/middleware/rateLimiter.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+const redis = require('../../config/redis')
+const redisClient = redis.redisClient
+const moment = require('moment')
+const WINDOW_SIZE_IN_HOURS = 24
+const MAX_WINDOW_REQUEST_COUNT = 100
+const WINDOW_LOG_INTERVAL_IN_HOURS = 1
+
+module.exports = {
+  customRateLimiter: (req, res, next) => {
```
[the remark this answers is not in the document]
Comment (Rupeshiya) [reply app/middleware/rateLimiter.js:9]: Rate limiter is working tried to send more than 100 requests then it's sending this responses:
image

[the remark this answers is not in the document]
Comment (Rupeshiya) [reply app/middleware/rateLimiter.js:9]: Increased no of requests in testing env to 500 and in development or prod 200 no of requests

**Comment by Rupeshiya** [comment]
**Rate limiter:**
rateLimiter

**Other test cases**
test

**Comment by Rupeshiya** [comment]
https://github.com/codeuino/social-platform-donut-backend/issues/137

**Review by vaibhavdaren — APPROVED** [review]

**Pull request merged**